src/pages/results.tsx

```diff
@@ -77,27 +80,78 @@ const MainContent = styled(Paper)`
   flex: 1 0 auto;
 `;
 
+const getParameters = <T extends Partial<FullSearchParameters>>(keys: (keyof T)[]) => {
```

Comment:
If `keys` is defined as type `keyof T`, do we need to filter `.filter(key => keys.includes(key as keyof T))`? This should theoretically enforce that `keys` are only defined to be keys of `Partial<FullSearchParameters>` and thus must be a valid key in `FullSearchParameters`.

Is it sufficient to instead do something like (on line 85)
```
keys.reduce((obj, key) => {
        obj[key] = fullSearchParams[key];
        return obj;
      }, {} as T);
```

Reply:
Update!

So because T only extends the Partial (and is not a Partial itself), it's not technically speaking guaranteed that a `keyof T` is a `keyof FullSearchParameters`. That being said, the following should work:

```
keys.reduce((obj, key) => {
        obj[key] = fullSearchParams[key as string];
        return obj;
      }, {} as T);
```
You need to cast the key as a string since T could add a symbol or number key that we don't know of. That being said, I still prefer this (looping over `keys`) over the filter method (looping over `fullSearchParams` AND `keys`) given the way we are utilizing the function. You can add in a quick check too to not add to `obj` if `fullSearchParams[key as string];` returns a null or something similar if desired.